File: Backend/PRScapstone/Models/PurchaseRequest.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace PRScapstone.Models
{
    public class PurchaseRequest
    {
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required][StringLength(100)]
        public string Description { get; set; }

        [Required][StringLength(255)]
        public string Justification { get; set; } //TODO date needed skipped/optional

        [Required][StringLength(25)]
        public string DeliveryMode { get; set; }

        // [Required]
        [StringLength(15)]
        public string Status { get; set; }

        [Required]
        public decimal Total { get; set; }

        [Required]
        public bool Active { get; set; } // TODO default true value?

        [StringLength(100)]
        public string ReasonForRejection { get; set; }

        // public DateTime? DateCreated { get; set; }

        // public DateTime? DateUpdated { get; set; }

        public virtual User User { get; set; }

        public virtual List<PurchaseRequestLineItem> PrliList { get; set; }

        public PurchaseRequest ()
        {
            Status = "NEW";
        }
    }
}

File: Frontend/src/app/models/purchaserequest.ts
import { User } from './user';
import { PRLI } from './prli';

export class PurchaseRequest {
  Id: number;
  UserId: number;
  Description: string;
  Justification: string;
  DeliveryMode: string;
  Status: string;
  Total: number;
  Active: boolean;
  ReasonForRejection: string;
  User: User;
  PrliList: PRLI[];

  constructor(
    Id: number,
    UserId: number,
    Description: string,
    Justification: string,
    DeliveryMode: string,
    Status: string,
    Total: number,
    Active: boolean,
    ReasonForRejection: string
  ) {
    this.Id = Id;
    this.UserId = UserId;
    this.Description = Description;
    this.Justification = Justification;
    this.DeliveryMode = DeliveryMode;
    this.Status = Status;
    this.Total = Total;
    this.Active = Active;
    this.ReasonForRejection = ReasonForRejection;
  }
}
